sa/model_test.go

```diff
@@ -224,3 +239,144 @@ func TestPopulateAttemptedFieldsBadJSON(t *testing.T) {
+	p := bigIntFromB64("uKE2dh-cTf6ERF4k4e_jy78GfPYUIaUyoSSJuBzp3Cubk3OCqs6grT8bR_cu0Dm1MZwWmtdqDyI95HrUeq3MP15vMMON8lHTeZu2lmKvwqW7anV5UzhM1iZ7z4yMkuUwFWoBvyY898EXvRD-hdqRxHlSqAZ192zB3pVFJ0s7pFc=")
+	q := bigIntFromB64("uKE2dh-cTf6ERF4k4e_jy78GfPYUIaUyoSSJuBzp3Cubk3OCqs6grT8bR_cu0Dm1MZwWmtdqDyI95HrUeq3MP15vMMON8lHTeZu2lmKvwqW7anV5UzhM1iZ7z4yMkuUwFWoBvyY898EXvRD-hdqRxHlSqAZ192zB3pVFJ0s7pFc=")
+	testKey := rsa.PrivateKey{PublicKey: rsa.PublicKey{N: n, E: e}, D: d, Primes: []*big.Int{p, q}}
+
+	certDer, _ := x509.CreateCertificate(rand.Reader, &rawCert, &rawCert, &testKey.PublicKey, &testKey)
+	cert := &core.Certificate{
+		RegistrationID: regID,
+		Serial:         serialString,
+		Expires:        rawCert.NotAfter,
+		DER:            certDer,
+	}
+	err := ctx.db.Insert(cert)
+	test.AssertNotError(t, err, "Couldn't insert certificate")
+}
+
+// setupCertificate1337A registers and inserts a record into the `certificates`
+// table with a domain of `1337.com`, `registrationID` of `1` and a `serial` of
+// `1337`.
+func (c testCtx) setupCertificate1337A(t *testing.T) {
+	jsonKey := []byte(`
+{
+  "kty":"RSA",
+  "n":"0vx7agoebGcQSuuPiLJXZptN9nndrQmbXEps2aiAFbWhM78LhWx4cbbfAAtVT86zwu1RK7aPFFxuhDR1L6tSoc_BJECPebWKRXjBZCiFV4n3oknjhMstn64tZ_2W-5JsGY4Hc5n9yBXArwl93lqt7_RN5w6Cf0h4QyQ5v-65YGjQR0_FDW2QvzqY368QQMicAtaSqzs8KJZgnYb9c7d0zgdAZHzu6qMQvRL5hajrn1n91CbOpbISD08qNLyrdkt-bFTWhAI4vMQFh6WeZu0fM4lFd2NcRwr3XPksINHaQ-G_xBniIqbw0Ls1jF44-csFCur-kEgU8awapJzKnqDKgw",
+  "e":"AQAB"
+}`)
+	c.registerCertificate(t, "test@example.com", jsonKey, 1)
+	c.insertCertificate(t, "1337.com", 1337, 1)
+}
+
+// setupCertificate1337A registers and inserts a record into the `certificates`
+// table with a domain of `1337.net`, `registrationID` of `1` and a `serial` of
+// `1337`.
+func (c testCtx) setupCertificate1337B(t *testing.T) {
+	jsonKey := []byte(`
+{
+  "kty":"RSA",
+  "n":"z8bp-jPtHt4lKBqepeKF28g_QAEOuEsCIou6sZ9ndsQsEjxEOQxQ0xNOQezsKa63eogw8YS3vzjUcPP5BJuVzfPfGd5NVUdT-vSSwxk3wvk_jtNqhrpcoG0elRPQfMVsQWmxCAXCVRz3xbcFI8GTe-syynG3l-g1IzYIIZVNI6jdljCZML1HOMTTW4f7uJJ8mM-08oQCeHbr5ejK7O2yMSSYxW03zY-Tj1iVEebROeMv6IEEJNFSS4yM-hLpNAqVuQxFGetwtwjDMC1Drs1dTWrPuUAAjKGrP151z1_dE74M5evpAhZUmpKv1hY-x85DC6N0hFPgowsanmTNNiV75w",
+  "e":"AQAB"
+}`)
+	c.registerCertificate(t, "test@example.com", jsonKey, 1)
+	c.insertCertificate(t, "1337.net", 1337, 1)
+}
+
+func setup(t *testing.T) testCtx {
+	dbMap, err := NewDbMap(vars.DBConnSAFullPerms, DbSettings{})
+	test.AssertNotError(t, err, "Couldn't create database connection")
+
+	cleanUp := test.ResetSATestDatabase(t)
+	fc := newFakeClock(t)
+
+	ssa, err := NewSQLStorageAuthority(dbMap, fc, log, metrics.NoopRegisterer, 1)
```

Comment:
The purpose of fake clocks is to be able to manipulate the time observed by our components (like SA). So the `fc` should become a field of `testCtx` so we can use it later.